### PR TITLE
executor: output to user-specified streams

### PR DESCRIPTION
**Pull request by cmatsuoka**
Output to user-specified stdout and stderr when executing a step. This
allows better control of application verbosity levels.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

CRAFT-832

### CODE REVIEW

**Comment by cmatsuoka** [comment]
Only merge this PR after PR #181 lands.

**Comment by cmatsuoka** [comment]
Rebased on main.

**Review by facundobatista — APPROVED** [review]
Looks great!

**Pull request merged**
